src/cmdlib.sh

```diff
@@ -416,11 +421,17 @@ runvm() {
                 ;;
         esac
     done
-    local vmpreparedir=${workdir}/tmp/supermin.prepare
-    local vmbuilddir=${workdir}/tmp/supermin.build
 
-    rm -rf "${vmpreparedir}" "${vmbuilddir}"
+    # shellcheck disable=SC2155
+    local runvm_tmp="$(mktemp --tmpdir="${workdir}/tmp" -d runvm.XXXX)"
+    local vmpreparedir="${runvm_tmp}/supermin.prepare"
+    local vmbuilddir="${runvm_tmp}/supermin.build"
+    local runvm_console="${runvm_tmp}/runvm-console.txt"
+    local rc_file="${runvm_tmp}/rc"
+
     mkdir -p "${vmpreparedir}" "${vmbuilddir}"
+    # shellcheck disable=SC2064
+    trap "rm -rf ${runvm_tmp}" EXIT
```

Comment:
Bash `trap`s literally are traps. :) When you set it, you overwrite the previous trap for that signal. And when a codebase gets large enough, it's not trivial to keep track of all the traps. In this case for example, I think this is overwriting the `trap` we've set in `buildextend-metal`, right?

We can avoid creating a separate tmpdir entirely here by just placing it in the `tmp_builddir`. See https://github.com/coreos/coreos-assembler/pull/1518/files (or really... feel free to take those bits and put them in this PR. Or drop the hunks for `runvm` and we can address it in a follow-up).

Reply:
Done. TBH, I was suspicous of doing the `trap` anyway, but that was before I worked out the `tmp_builddir` change. I've rolled in some of the changes from !1518 (osmet, and cribbed off .cci.jenkins).